src/main/webapp/js/components/interface/drawer/TabbedDrawer.js

```diff
@@ -79,90 +78,117 @@ define(function (require) {
                 );
             }, this);
             return renderedLabels;
-        }
+        };
 
         // function to render all the childs, wrap each one of them in a div and manage with display
         // which child has to be visible.
-        renderMyChildren() {
-            var renderedComponents = this.props.children.map(function(child, index) {
-                if(this.state.drawerOpened == true) {
-                    var ElementToRender = child;
-                    var MyElement = (this.state.selectedTab != index) ? <div className="hiddenComponent" key={index}><ElementToRender /></div> : <div key={index}><ElementToRender /></div>;
-
-                } else {
-                    var ElementToRender = child;
-                    var MyElement = <div key={index}><ElementToRender className="hiddenComponent" /></div>;
-                }
-                return (MyElement);
-            }, this);
-            return renderedComponents;
-        }
+        renderMyChildren = () => {
+            var paddingChildren = 45;
+            if (this.props.children != undefined) {
+                var renderedComponents = this.props.children.map(function (child, index) {
+                    var ElementToRender = undefined;
+                    var MyElement = undefined;
+                    if(child.type != undefined) {
+                        var ElementToRender = child.type;
+                        if (this.state.drawerOpened == true) {
```

Comment:
It looks to me the different between the code inside the `if` and the `else` is just that in one of them we are using `{...child.props}` while in the second one we use `{...this.props.childrenProps[index]}`. Maybe we can assign that to a variable and keep the _html code_ in a single place, something along these lines:
```
if(child.type != undefined) {
  var ElementToRender = child.type;
  var properties = child.props;
}
else{
var ElementToRender = child;
var properties = this.props.childrenProps[index]
}
if (this.state.drawerOpened == true) {
  MyElement = (this.state.selectedTab != index) ? <div className="hiddenComponent" key={index}><ElementToRender iframeHeight={this.state.drawerHeight - paddingChildren} {...properties} /></div> : <div key={index}><ElementToRender iframeHeight={this.state.drawerHeight - paddingChildren} {...properties} /></div>;
} else {
   MyElement = <div key={index}><ElementToRender className="hiddenComponent" iframeHeight={this.state.drawerHeight - paddingChildren} {...properties} /></div>;
}
```

Reply:
Good point, will do, thx.

Reply:
I have just made a final commit that polish a bit your refactor. Can you please double check with my commit and that it doesn't break anything nor in netpyne neither with the default extension (adding the children with the component factory)? If you are happy with it I will approve the PR